docs/vspec2x_arch.md

```diff
@@ -22,6 +19,28 @@ This will result in that e.g. `Vehicle.Cabin.Door` is expanded to the following
 For some exporters expansion can be suppressed by using the `--no_expand` option.
 Then instance information will be represented by other means in the resulting output.
 
+## Deletion / Node removal
+
+Nodes can be removed from the tree by using the `delete` keyword in the overlay.
+This is useful when a signal is not used in a next version of the specification or if you
+simply want to delete a node or branch from the specification, e.g. you are not using a signal of the base
+specification.
+Please note, that the deletion for branches will delete all nodes that are connected to that branch (which is
+desired behavior).
+
+If you want to delete a node or branch simply add the following to the specification/overlay, let's say the vehicle only
+has two doors in the front. In this case we would like to delete the signals for the rear doors:
+
+```
+Vehicle.Cabin.Door.Row2:
```

Comment:
This example is possibly incorrect as it refers to a Row2 instance, which does not exist, per se, in the catalog. You may need a better example. Or, is it intentional that the deletion would support deleting specific instances? If so, I am not sure why the instance would be created in the first place...

As well, this raises a very interesting issue for which I will raise a separate issue or discussion point related to the MPL-2.0 license. Section 1.10b states that the license covers: "any new file in Source Code Form that contains any Covered Software". Given that one must provide in the overlay file the node path name of the node to be deleted, the license requires that this overlay file be made available upon request and that the license terms apply to this new file. The same applies to an overlay that modifies an existing signal (say, to change the unit or set a different number of instances).

This defeats the goals of providing the overlay mechanism to allow third-party implementors of a catalog to create their own version independent of the VSS version and license. We probably need a broader discussion of the license.

Indeed, the text at https://covesa.github.io/vehicle_signal_specification/license/ mentions that deletions are not considered to run afoul of the license but these statements are external to the license itself and, even if COVESA has no intention to enforce this part of the license, it remains that lawyers are generally worried by such statements. It's generally preferable to address this in the license itself. Anyway, a separate discussion to this PR, to be continued once I open a separate issue or discussion.

Reply:
> This example is possibly incorrect as it refers to a Row2 instance, which does not exist, per se, in the catalog. You may need a better example. Or, is it intentional that the deletion would support deleting specific instances? If so, I am not sure why the instance would be created in the first place...

Up for discussion, to me it seemed like a good example to show that even deleting instances would work if `delete` is used from overlay, but maybe you are right and I should split it to show multiple examples:
I would add **one example for deleting a node**, **one for deleting a branch** and should we add **one for the instances** or how does deleting instances work currently? Maybe I'm missing some information here, could you explain? Let's say a car only has only one row instead of two, would this be removed from the base `VehicleSignalSpecification.vspec`?

Reply:
@ppb2020 - concerning the license topic - there have been some previous discussions on license for VSS as some seem to prefer for example Apache over MPL due to concerns similar to yours. It is anyhow a separate discussion, check with @paulboyes if you want to discuss it further.

Reply:
> @ppb2020 - concerning the license topic - there have been some previous discussions on license for VSS as some seem to prefer for example Apache over MPL due to concerns similar to yours. It is anyhow a separate discussion, check with @paulboyes if you want to discuss it further.

Wilco.